File: frontend/src/apis/setting.ts
import { request } from "../utils/requests";

export async function getSettings() {
    const response = await request.request({
        url: '/setting/',
    })
    return response.data.data
}

export function saveSetting(section: string, data: any) {
    return request.request({
        url: '/setting/',
        method: 'post',
        params: { section },
        data: data
    })
}

/**
 * 测试qBittorrent下载器连接
 */
export function testQBittorrentConnection() {
    return request.request({
        url: '/setting/test-qbittorrent',
        method: 'get'
    })
}

File: app/api/setting.py
"""
Author: Await
Date: 2025-05-24 17:05:38
LastEditors: Await
LastEditTime: 2025-05-27 14:27:14
Description: 请填写简介
"""

from fastapi import APIRouter

from app.integrations.downloaders.manager import downloader_manager
from app.schema import Setting
from app.schema.r import R
from app.service.setting import SettingService

router = APIRouter()


@router.get("/")
def get_settings():
    settings = Setting()
    return R.ok(settings)


@router.post("/")
def save_setting(section: str, setting: dict):
    SettingService.save_section(section, setting)
    return R.ok()


@router.get("/test-qbittorrent")
def test_qbittorrent_connection():
    result = downloader_manager.get_active().test_connection()
    if result["status"]:
        return R.ok(result)
    return R.fail(result["message"], data=result)
